### DISCUSSION TITLE
wasi as common language runtime

### DISCUSSION THREAD

**Discussion by kesavkolla** [duckdb/duckdb#5096] [Ideas]
Today duckdb has several lagnuage integrations like Python, R, Ruby etc... I think it will be tough keep up with language needs as no.of programming languages grow. Like for example there is no language bindings for Rust today. I think instead of supporting individual programming languages is it possible to support WASI using wasmer or wasmtime? That way the language support can reach to several programming languages.

Infact the extension mechanism can also benefit a lot with WASI so that people can develop extensions in their programming language of choice.

IMO both extension and individual language bindings should be done using WASI.

**Comment by hannes** [comment]
The runtime is the easy part, the hard part of the language bindings is integrating with the environment, e.g. with Pandas on Python. Having WASM there only solves the easy part.

**Reply by kesavkolla** [reply]
Agree... Maybe extensions would benefit from having wasm.
